code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -406,42 +406,46 @@
 	togglelock(user)
 
 /mob/living/silicon/robot/attackby(obj/item/W, mob/user, params)
+	if(length(user.progressbars))
+		if(W.tool_behaviour == TOOL_WELDER || istype(W, /obj/item/stack/cable_coil))
+			user.changeNext_move(CLICK_CD_MELEE)
+			to_chat(user, "<span class='notice'>You are already busy!</span>")
+			return
 	if(W.tool_behaviour == TOOL_WELDER && (user.a_intent != INTENT_HARM || user == src))
 		user.changeNext_move(CLICK_CD_MELEE)
 		if (!getBruteLoss())
 			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
 			return
-		if (!W.tool_start_check(user, amount=0)) //The welder has 1u of fuel consumed by it's afterattack, so we don't need to worry about taking any away.
-			return
-		if(src == user)
-			to_chat(user, "<span class='notice'>You start fixing yourself.</span>")
-			if(!W.use_tool(src, user, 50))
+		while (getBruteLoss() && W.tool_start_check(user, amount=0))	//Repeatedly attempt to repair dents until done or welder is out of fuel, just like tend wounds. No need to spam click
+			if(src == user)
+				to_chat(user, "<span class='notice'>You start fixing yourself.</span>")
+			if(!W.use_tool(src, user, 60))
 				return
```

Comment:
This should probably be a condition of the loop

Reply:
👍 

Did for both welder and cables